### PR TITLE
A few more unit tests

### PR DESCRIPTION
**Pull request by lightglitch**
Don't know what is the best option to test the private methods of DefaultExtensionFinder ...

### CODE REVIEW

**Comment by decebals** [comment]
I am so happy when I see a new unit test :smile: I try to learn as much as possible from this project. I am sure that we must modified (a little bit) the project code to facilitate the creation of unit tests. 
Also I think that we must use (as much as possible) immutable (not changeable) classes with Builder pattern for instantiation to facilitate the thread safe implementation.
I will come with comments at this PR asap. Thanks!

**Comment by lightglitch** [comment]
My experience in unit testing is also not very high so any suggestions are welcome.

File: pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java [hunk 1]
```diff
@@ -51,10 +51,10 @@ public DefaultExtensionFinder(PluginManager pluginManager, ExtensionFactory exte
         }
 
 		log.debug("Finding extensions for extension point '{}'", type.getName());
-        readIndexFiles();
+        Map<String, Set<String>> cachedEntries = readIndexFiles();
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java]: I don't understand this modification. What was wrong in the previous version?

File: pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ */
+package ro.fortsoft.pf4j.plugin;
+
+import ro.fortsoft.pf4j.ExtensionPoint;
+
+/**
+ *
+ * @author Mario Franco
+ */
+public interface TestExtensionInterface extends ExtensionPoint {
```
Comment (decebals) [inline pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java]: Maybe `TestExtensionPoint` is a better name?

File: pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java (hunk 1 above)
Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java]: It was to make it easy to test but if you have another suggestion I can revert this.

File: pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java (hunk 2 above)
Comment (lightglitch) [inline pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java]: Yep.

File: pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java (hunk 1 above)
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultExtensionFinder.java]: Can you do a rebase please. I modified the class a little bit to improve the testability.

**Comment by decebals** [comment]
> Don't know what is the best option to test the private methods of DefaultExtensionFinder ...

Maybe this [link](http://www.artima.com/suiterunner/private.html) help.

**Comment by janhoy** [comment]
Just browsing this project to consider it for use. Why is this and two other PR's still open since 2015? I would expect them to be either merged or closed?

**Comment by decebals** [comment]
@janhoy 
Maybe, because the code is not ready for merge (are some comments without response).
For example I see in one comment:
> Can you do a rebase please. I modified the class a little bit to improve the testability.
These PR are not closed because they comes with a big value (improve testability) and I hope to merge them in the future.

Right now for me it's important to merge https://github.com/decebals/pf4j/pull/128, because this PR comes with useful things and it contains some refactored code. After I release the next version (in few days), I will try to take a look over these old PR/issues and to come with a resolution.

**Comment by janhoy** [comment]
+1

**Comment by lightglitch** [comment]
I have been very busy but I will try to update the pull request soon.

File: pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFinderTest.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2015 Mario Franco.
```
Comment (decebals) [inline pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFinderTest.java]: Please change the copyright (they are some files in this PR).

File: pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java (hunk 2 above)
Comment (decebals) [inline pf4j/src/test/java/ro/fortsoft/pf4j/plugin/TestExtensionInterface.java]: Please rename it to `TestExtensionPoint` (see above comments).

**Comment by lightglitch** [comment]
The tests need to be updated I will tell you when it's OK to merge it

**Comment by decebals** [comment]
@lightglitch I was the impression that you made some changes :smile:

**Comment by decebals** [comment]
@lightglitch With your permission, I will integrate these tests in master.

**Comment by decebals** [comment]
Resolved by 0a0513b6634cf7ec4689a2ef88862f4d885f73c8.
@lightglitch Thanks!

**Pull request closed without merging**